.github/workflows/ci-all.yml

```diff
@@ -0,0 +1,80 @@
+name: CI - All
+
+on:
+  push:
+    branches: ["main"]
+    paths:
+      - "host/**"
+      - "lib/**"
+      - "primitives/**"
+      - "script/**"
+  pull_request:
+    paths:
+      - "host/**"
+      - "lib/**"
+      - "primitives/**"
+      - "script/**"
+env:
+  CARGO_TERM_COLOR: always
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build-test-native:
+    uses: ./.github/workflows/ci-native.yml
+
+  build-test-risc0:
+    uses: ./.github/workflows/ci-risc0.yml
+
+  build-test-sp1:
+    uses: ./.github/workflows/ci-sp1.yml
+
+  build-test-sgx-all:
+    uses: ./.github/workflows/ci-sgx-all.yml
+
+  test-lib:
+    name: Test raiko-lib
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+
+      - run: cargo test -p raiko-lib --features=std
+
+  clippy:
```

Comment:
hmm i think all rust code should get clippy and fmt. can we pull these lint stuff out into a ci-lint.yml and maybe call that on any PR? then it really logically makes sense to have ci-all and ci-zkVM